app/models/post.rb

```diff
@@ -77,7 +77,9 @@ class Post < ApplicationRecord
     select('(SELECT COUNT(*) FROM replies WHERE replies.post_id = posts.id) AS reply_count')
   }
 
-  def visible_to?(user)
+  def visible_to?(user, show_blocked=false)
+    return false if !show_blocked && user&.author_hidden?(self, author_ids)
```

Comment:
authorship (and authors_locked below) seems distinct from if the post is viewable at all; this seems like two mixed concepts